--- origsrc/abiword-2.6.4/ac-helpers/abi-detect-platform.m4	2008-07-13 08:09:15.000000000 -0500
+++ src/abiword-2.6.4/ac-helpers/abi-detect-platform.m4	2008-07-16 16:22:02.500000000 -0500
@@ -43,26 +43,6 @@
 
 ##################################################################
 ##################################################################
-#### Cygnus keeps changing the value that uname returns between
-#### different versions of the package and between different
-#### versions of Windows.  Here we fold them all in into one symbol.
-
-if test "$OS_NAME" = "CYGWIN32"; then
-	OS_NAME="WIN32"
-fi
-
-if test "$OS_NAME" = "CYGWIN"; then
-	OS_NAME="WIN32"
-fi
-
-if test "$OS_NAME" = "WIN32"; then
-	CYGWIN_MAJOR_VERSION=`echo $(OS_RELEASE) | cut -d . -f 1`
-	CYGWIN_MINOR_VERSION=`echo $(OS_RELEASE) | cut -d . -f 2`
-	CYGWIN_REVISION=`echo $(OS_RELEASE) | cut -d . -f 3`
-fi
-
-##################################################################
-##################################################################
 #### MINGW configuration
 
 if test "$OS_NAME" = "MINGW32"; then
@@ -182,6 +162,9 @@
 dnl why not call AC_CANONICAL_HOST or AC_CANONICAL_SYSTEM and
 dnl use those values instead?
 case "$OS_NAME" in 
+	CYGWIN)
+		WARNING_CFLAGS="-Wall -D_BSD_SOURCE -pipe"
+		;;
 	MACOSX)
 		# the default for MacOS X is to no enable warnings with
 		# the Apple-derived gcc.
@@ -265,7 +248,7 @@
 	WIN32)
 		PLATFORM="win"
 		;;
-	Linux|AIX|*BSD|IRIX*|HP|OSF1|SunOS|DragonFly)
+	Linux|AIX|*BSD|IRIX*|HP|OSF1|SunOS|DragonFly|CYGWIN)
 		PLATFORM="unix"
 		;;
 	QNX|procnto)
