--- origsrc/abiword-2.6.4/ac-helpers/abi-lib.m4	2008-07-13 08:09:16.000000000 -0500
+++ src/abiword-2.6.4/ac-helpers/abi-lib.m4	2008-07-16 16:22:02.546875000 -0500
@@ -38,11 +38,20 @@
 ])
 
 if test "$libabiword" = true ; then
-	CFLAGS="$CFLAGS -fPIC"
-	CXXFLAGS="$CXXFLAGS -fPIC"
+	if test "$OS_NAME" = CYGWIN ; then
+		ABIWORD_LIBRARY="libabiword-\$(ABIWORD_SERIES).dll.a"
+		ABIWORD_DLL="cygabiword-\$(ABIWORD_SERIES).dll"
+	else
+		ABIWORD_LIBRARY="libabiword-\$(ABIWORD_SERIES).so"
+		CFLAGS="$CFLAGS -fPIC"
+		CXXFLAGS="$CXXFLAGS -fPIC"
+	fi
 fi
 
+AC_SUBST(ABIWORD_LIBRARY)
+AC_SUBST(ABIWORD_DLL)
 AM_CONDITIONAL(WITH_LIBABIWORD, test "x$libabiword" = "xtrue")
+AM_CONDITIONAL(WITH_LIBABIWORD_DLL, test "x$ABIWORD_DLL" != "x")
 
 ])
 # 
--- origsrc/abiword-2.6.4/src/wp/main/unix/GNUmakefile.am	2008-07-13 08:10:40.000000000 -0500
+++ src/abiword-2.6.4/src/wp/main/unix/GNUmakefile.am	2008-07-16 16:21:47.218750000 -0500
@@ -46,11 +46,10 @@
 headers_DATA = \
 	libabiword.h
 
-ABIWORD_LIBRARY = libabiword-$(ABIWORD_SERIES).so
 ABIWORD_SODIR = $(shell pwd)
-ABIWORD_DYNBINARY = abiword-dynamic
+ABIWORD_DYNBINARY = abiword$(EXEEXT)
 
-$(ABIWORD_LIBRARY): $(ABI_LIBS) $(srcdir)/libabiword.cpp $(srcdir)/../xp/abi_ver.cpp
+libabiword-$(ABIWORD_SERIES).so: $(ABI_LIBS) $(srcdir)/libabiword.cpp $(srcdir)/../xp/abi_ver.cpp
 	$(CXX) -shared -Wl,-soname,$(ABIWORD_LIBRARY) -o $@ 		\
 		$(CPPFLAGS) $(CXXFLAGS) $(INCLUDES)			\
 		-DPACKAGE="\"$(PACKAGE)\""				\
@@ -61,6 +60,17 @@
 		-Wl,--export-dynamic					\
 		-Wl,--no-undefined
 
+libabiword-$(ABIWORD_SERIES).dll.a: $(ABI_LIBS) $(srcdir)/libabiword.cpp $(srcdir)/../xp/abi_ver.cpp
+	$(CXX) -shared -Wl,--out-implib,$@ -o $(ABIWORD_DLL) 		\
+		$(CPPFLAGS) $(CXXFLAGS) $(INCLUDES)			\
+		-DPACKAGE="\"$(PACKAGE)\""				\
+		$(srcdir)/../xp/abi_ver.cpp				\
+		$(srcdir)/libabiword.cpp				\
+		-Wl,--whole-archive $(ABI_LIBS) 			\
+		-Wl,--no-whole-archive $(PLATFORM_LIBS) $(OTHER_LIBS)	\
+		-Wl,--export-dynamic					\
+		-Wl,--no-undefined
+
 $(ABIWORD_DYNBINARY): $(ABIWORD_LIBRARY) $(srcdir)/UnixMain.cpp
 	$(CXX) -o $@					\
 		$(CPPFLAGS) $(CXXFLAGS) $(INCLUDES) 	\
@@ -76,6 +86,10 @@
 install-libabiword: 
 	@mkdir -p $(DESTDIR)/$(libdir)
 	cp $(ABIWORD_LIBRARY) $(DESTDIR)/$(libdir)/$(ABIWORD_LIBRARY)
+if WITH_LIBABIWORD_DLL
+	@mkdir -p $(DESTDIR)/$(bindir)
+	cp $(ABIWORD_DLL) $(DESTDIR)/$(bindir)/$(ABIWORD_DLL)
+endif
 
 endif
 
