--- origsrc/abiword-2.6.4/src/wp/ap/unix/ap_UnixApp.cpp	2008-07-13 08:10:00.000000000 -0500
+++ src/abiword-2.6.4/src/wp/ap/unix/ap_UnixApp.cpp	2008-07-16 16:22:02.656250000 -0500
@@ -778,10 +778,11 @@
 		if ( name )
 		{
 			int len = strlen (name);
+			int ext = strlen ("."G_MODULE_SUFFIX);
 			
-			if (len >= 3)
+			if (len >= ext)
 			{
-				if(!strcmp(name+(len-3), "."G_MODULE_SUFFIX))
+				if(!strcmp(name+(len-ext), "."G_MODULE_SUFFIX))
 					return 1;
 			}
 		}
@@ -837,13 +838,14 @@
 			  UT_DEBUGMSG(("DOM: loading plugin %s\n", plugin.c_str()));
 
 			  int len = strlen (namelist[n]->d_name);
-			  if (len < 4)
+			  int ext = strlen ("."G_MODULE_SUFFIX);
+			  if (len < ext + 1)
 			  {
 				  UT_DEBUGMSG(("FJF: bad name for a plugin\n"));
 				  g_free(namelist[n]);
 				  continue;
 			  }
-			  if(strcmp (namelist[n]->d_name+(len-3), "."G_MODULE_SUFFIX) != 0)
+			  if(strcmp (namelist[n]->d_name+(len-ext), "."G_MODULE_SUFFIX) != 0)
 			  {
 				  UT_DEBUGMSG(("FJF: not really a plugin?\n"));
 				  g_free(namelist[n]);
