--- origsrc/abiword-2.8.0/src/Makefile.am	2009-09-09 08:11:01.000000000 -0500
+++ src/abiword-2.8.0/src/Makefile.am	2009-10-28 19:19:31.222430400 -0500
@@ -44,7 +44,7 @@ libabiword_2_8_la_CPPFLAGS = \
 
 libabiword_2_8_la_LDFLAGS = \
 	$(DEPS_LIBS) \
-	--no-undefined \
+	-no-undefined \
 	-avoid-version \
 	-export-dynamic
 
@@ -107,7 +107,7 @@ AbiWord_CPPFLAGS = \
 
 AbiWord_LDFLAGS = \
 	$(DEPS_LIBS) \
-	--no-undefined \
+	-no-undefined \
 	-avoid-version \
 	-export-dynamic \
 	-headerpad_max_install_names 
