--- origsrc/abiword-2.9.4/src/text/ptbl/xp/pd_DocumentRDF.cpp	2012-10-02 09:15:07.000000000 -0500
+++ src/abiword-2.9.4/src/text/ptbl/xp/pd_DocumentRDF.cpp	2012-12-27 02:29:51.681061500 -0600
@@ -1884,7 +1884,7 @@ PD_RDFContact::className() const
 
 #ifdef WITH_EVOLUTION_DATA_SERVER
 extern "C" {
-  #include <libebook/e-book.h>
+  #include <libebook/libebook.h>
 };
 
 static std::string get( EVCard* c, const char* v )
