--- origsrc/abiword-2.6.4/GNUmakefile.am	2008-07-13 08:11:02.000000000 -0500
+++ src/abiword-2.6.4/GNUmakefile.am	2008-07-16 16:22:02.468750000 -0500
@@ -25,15 +25,21 @@
 
 if WITH_LIBABIWORD
 pc_template = libabiword.pc
+pc_uninst_template = libabiword-uninstalled.pc
 else
 pc_template = abiword.pc
+pc_uninst_template = $(pc_template)
 endif
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = abiword-$(ABIWORD_SERIES).pc
+noinst_DATA = abiword-$(ABIWORD_SERIES)-uninstalled.pc
 
 abiword-$(ABIWORD_SERIES).pc: $(pc_template)
 	cp $< abiword-$(ABIWORD_SERIES).pc
 
+abiword-$(ABIWORD_SERIES)-uninstalled.pc: $(pc_uninst_template)
+	cp $< $@
+
 EXTRA_DIST = 			\
 	docs			\
 	po			\
--- origsrc/abiword-2.6.4/ac-helpers/abi-detect-platform.m4	2008-07-13 08:09:15.000000000 -0500
+++ src/abiword-2.6.4/ac-helpers/abi-detect-platform.m4	2008-07-16 16:22:02.500000000 -0500
@@ -43,26 +43,6 @@
 
 ##################################################################
 ##################################################################
-#### Cygnus keeps changing the value that uname returns between
-#### different versions of the package and between different
-#### versions of Windows.  Here we fold them all in into one symbol.
-
-if test "$OS_NAME" = "CYGWIN32"; then
-	OS_NAME="WIN32"
-fi
-
-if test "$OS_NAME" = "CYGWIN"; then
-	OS_NAME="WIN32"
-fi
-
-if test "$OS_NAME" = "WIN32"; then
-	CYGWIN_MAJOR_VERSION=`echo $(OS_RELEASE) | cut -d . -f 1`
-	CYGWIN_MINOR_VERSION=`echo $(OS_RELEASE) | cut -d . -f 2`
-	CYGWIN_REVISION=`echo $(OS_RELEASE) | cut -d . -f 3`
-fi
-
-##################################################################
-##################################################################
 #### MINGW configuration
 
 if test "$OS_NAME" = "MINGW32"; then
@@ -182,6 +162,9 @@
 dnl why not call AC_CANONICAL_HOST or AC_CANONICAL_SYSTEM and
 dnl use those values instead?
 case "$OS_NAME" in 
+	CYGWIN)
+		WARNING_CFLAGS="-Wall -D_BSD_SOURCE -pipe"
+		;;
 	MACOSX)
 		# the default for MacOS X is to no enable warnings with
 		# the Apple-derived gcc.
@@ -265,7 +248,7 @@
 	WIN32)
 		PLATFORM="win"
 		;;
-	Linux|AIX|*BSD|IRIX*|HP|OSF1|SunOS|DragonFly)
+	Linux|AIX|*BSD|IRIX*|HP|OSF1|SunOS|DragonFly|CYGWIN)
 		PLATFORM="unix"
 		;;
 	QNX|procnto)
--- origsrc/abiword-2.6.4/ac-helpers/abi-lib.m4	2008-07-13 08:09:16.000000000 -0500
+++ src/abiword-2.6.4/ac-helpers/abi-lib.m4	2008-07-16 16:22:02.546875000 -0500
@@ -38,11 +38,20 @@
 ])
 
 if test "$libabiword" = true ; then
-	CFLAGS="$CFLAGS -fPIC"
-	CXXFLAGS="$CXXFLAGS -fPIC"
+	if test "$OS_NAME" = CYGWIN ; then
+		ABIWORD_LIBRARY="libabiword-\$(ABIWORD_SERIES).dll.a"
+		ABIWORD_DLL="cygabiword-\$(ABIWORD_SERIES).dll"
+	else
+		ABIWORD_LIBRARY="libabiword-\$(ABIWORD_SERIES).so"
+		CFLAGS="$CFLAGS -fPIC"
+		CXXFLAGS="$CXXFLAGS -fPIC"
+	fi
 fi
 
+AC_SUBST(ABIWORD_LIBRARY)
+AC_SUBST(ABIWORD_DLL)
 AM_CONDITIONAL(WITH_LIBABIWORD, test "x$libabiword" = "xtrue")
+AM_CONDITIONAL(WITH_LIBABIWORD_DLL, test "x$ABIWORD_DLL" != "x")
 
 ])
 # 
--- origsrc/abiword-2.6.4/configure.ac	2008-07-13 08:11:02.000000000 -0500
+++ src/abiword-2.6.4/configure.ac	2008-07-16 16:22:02.578125000 -0500
@@ -487,6 +487,7 @@
 AC_CONFIG_FILES([
  abiword.pc
  libabiword.pc
+ libabiword-uninstalled.pc
  GNUmakefile
  src/GNUmakefile
  src/af/GNUmakefile
--- origsrc/abiword-2.6.4/libabiword-uninstalled.pc.in	1969-12-31 18:00:00.000000000 -0600
+++ src/abiword-2.6.4/libabiword-uninstalled.pc.in	2008-07-16 16:22:02.625000000 -0500
@@ -0,0 +1,9 @@
+datarootdir=@datarootdir@
+pkgdatadir=@ABIWORD_DATADIR@
+libabiword=true
+
+Name: abiword-@ABIWORD_SERIES@
+Description: AbiWord word processing library
+Version: @VERSION@
+Requires: @ABIWORD_REQUIRED_PKGS@
+Libs: -L${pcfiledir}/src/wp/main/@PLATFORM@ -labiword-@ABIWORD_SERIES@
--- origsrc/abiword-2.6.4/src/wp/ap/unix/ap_UnixApp.cpp	2008-07-13 08:10:00.000000000 -0500
+++ src/abiword-2.6.4/src/wp/ap/unix/ap_UnixApp.cpp	2008-07-16 16:22:02.656250000 -0500
@@ -778,10 +778,11 @@
 		if ( name )
 		{
 			int len = strlen (name);
+			int ext = strlen ("."G_MODULE_SUFFIX);
 			
-			if (len >= 3)
+			if (len >= ext)
 			{
-				if(!strcmp(name+(len-3), "."G_MODULE_SUFFIX))
+				if(!strcmp(name+(len-ext), "."G_MODULE_SUFFIX))
 					return 1;
 			}
 		}
@@ -837,13 +838,14 @@
 			  UT_DEBUGMSG(("DOM: loading plugin %s\n", plugin.c_str()));
 
 			  int len = strlen (namelist[n]->d_name);
-			  if (len < 4)
+			  int ext = strlen ("."G_MODULE_SUFFIX);
+			  if (len < ext + 1)
 			  {
 				  UT_DEBUGMSG(("FJF: bad name for a plugin\n"));
 				  g_free(namelist[n]);
 				  continue;
 			  }
-			  if(strcmp (namelist[n]->d_name+(len-3), "."G_MODULE_SUFFIX) != 0)
+			  if(strcmp (namelist[n]->d_name+(len-ext), "."G_MODULE_SUFFIX) != 0)
 			  {
 				  UT_DEBUGMSG(("FJF: not really a plugin?\n"));
 				  g_free(namelist[n]);
--- origsrc/abiword-2.6.4/src/wp/main/unix/GNUmakefile.am	2008-07-13 08:10:40.000000000 -0500
+++ src/abiword-2.6.4/src/wp/main/unix/GNUmakefile.am	2008-07-16 16:21:47.218750000 -0500
@@ -46,11 +46,10 @@
 headers_DATA = \
 	libabiword.h
 
-ABIWORD_LIBRARY = libabiword-$(ABIWORD_SERIES).so
 ABIWORD_SODIR = $(shell pwd)
-ABIWORD_DYNBINARY = abiword-dynamic
+ABIWORD_DYNBINARY = abiword$(EXEEXT)
 
-$(ABIWORD_LIBRARY): $(ABI_LIBS) $(srcdir)/libabiword.cpp $(srcdir)/../xp/abi_ver.cpp
+libabiword-$(ABIWORD_SERIES).so: $(ABI_LIBS) $(srcdir)/libabiword.cpp $(srcdir)/../xp/abi_ver.cpp
 	$(CXX) -shared -Wl,-soname,$(ABIWORD_LIBRARY) -o $@ 		\
 		$(CPPFLAGS) $(CXXFLAGS) $(INCLUDES)			\
 		-DPACKAGE="\"$(PACKAGE)\""				\
@@ -61,6 +60,17 @@
 		-Wl,--export-dynamic					\
 		-Wl,--no-undefined
 
+libabiword-$(ABIWORD_SERIES).dll.a: $(ABI_LIBS) $(srcdir)/libabiword.cpp $(srcdir)/../xp/abi_ver.cpp
+	$(CXX) -shared -Wl,--out-implib,$@ -o $(ABIWORD_DLL) 		\
+		$(CPPFLAGS) $(CXXFLAGS) $(INCLUDES)			\
+		-DPACKAGE="\"$(PACKAGE)\""				\
+		$(srcdir)/../xp/abi_ver.cpp				\
+		$(srcdir)/libabiword.cpp				\
+		-Wl,--whole-archive $(ABI_LIBS) 			\
+		-Wl,--no-whole-archive $(PLATFORM_LIBS) $(OTHER_LIBS)	\
+		-Wl,--export-dynamic					\
+		-Wl,--no-undefined
+
 $(ABIWORD_DYNBINARY): $(ABIWORD_LIBRARY) $(srcdir)/UnixMain.cpp
 	$(CXX) -o $@					\
 		$(CPPFLAGS) $(CXXFLAGS) $(INCLUDES) 	\
@@ -76,6 +86,10 @@
 install-libabiword: 
 	@mkdir -p $(DESTDIR)/$(libdir)
 	cp $(ABIWORD_LIBRARY) $(DESTDIR)/$(libdir)/$(ABIWORD_LIBRARY)
+if WITH_LIBABIWORD_DLL
+	@mkdir -p $(DESTDIR)/$(bindir)
+	cp $(ABIWORD_DLL) $(DESTDIR)/$(bindir)/$(ABIWORD_DLL)
+endif
 
 endif
 
--- origsrc/abiword-plugins-2.6.4/ac-helpers/abi-plugin-platform.m4	2008-07-13 07:05:25.000000000 -0500
+++ src/abiword-plugins-2.6.4/ac-helpers/abi-plugin-platform.m4	2008-07-16 16:22:02.687500000 -0500
@@ -162,7 +162,7 @@
 		libgnomeprintui-2.2 >= 2.2.1
 	])
 	PKG_CHECK_MODULES(NAUTILUS,[
-		libnautilus >= 2.0
+		libnautilus-extension >= 2.0
 	],[	_abi_gnome_nautilus=yes
 	],[	_abi_gnome_nautilus=no
 	])
--- origsrc/abiword-plugins-2.6.4/tools/abicommand/plugin.m4	2008-07-13 07:04:53.000000000 -0500
+++ src/abiword-plugins-2.6.4/tools/abicommand/plugin.m4	2008-07-16 16:22:02.734375000 -0500
@@ -40,16 +40,16 @@
 have_readline=unknown
 AC_CHECK_HEADER(readline/readline.h,[
 	AC_CHECK_HEADER(readline/history.h,[
-		AC_CHECK_LIB(readline,readline,[
+		AC_CHECK_LIB(readline,rl_initialize,[
 			have_readline=yes
-			LDFLAGS="-ltermcap $LDFLAGS"
-			ABI_COMMAND_LDFLAGS="-ltermcap $ABI_COMMAND_LDFLAGS"
-		],[	AC_CHECK_LIB(readline,rl_initialize,[
+			LDFLAGS="-lcurses $LDFLAGS"
+			ABI_COMMAND_LDFLAGS="-lcurses $ABI_COMMAND_LDFLAGS"
+		],[	AC_CHECK_LIB(readline,readline,[
 				have_readline=yes
-				LDFLAGS="-lcurses $LDFLAGS"
-				ABI_COMMAND_LDFLAGS="-lcurses $ABI_COMMAND_LDFLAGS"
-			],have_readline=no,-lcurses)
-		],-ltermcap)
+				LDFLAGS="-ltermcap $LDFLAGS"
+				ABI_COMMAND_LDFLAGS="-ltermcap $ABI_COMMAND_LDFLAGS"
+			],have_readline=no,-ltermcap)
+		],-lcurses)
 	],have_readline=no)
 ],have_readline=no)
 
--- origsrc/abiword-plugins-2.6.4/tools/abigoffice/unix/GNUmakefile.am	2008-07-13 07:05:23.000000000 -0500
+++ src/abiword-plugins-2.6.4/tools/abigoffice/unix/GNUmakefile.am	2008-07-16 16:22:02.765625000 -0500
@@ -41,6 +41,8 @@
 
 libAbiGOffice_la_LDFLAGS = -no-undefined -module -avoid-version
 
+libAbiGOffice_la_LIBADD = @ABI_LIBS@
+
 # remove libAbiGOChart if installed
 install-data-hook:
 	if [ -x @ABI_PLUGIN_DIR@/libAbiGOChart.so ]; \
--- origsrc/abiword-plugins-2.6.4/tools/gda/plugin.m4	2008-07-13 07:04:54.000000000 -0500
+++ src/abiword-plugins-2.6.4/tools/gda/plugin.m4	2008-07-16 16:22:02.812500000 -0500
@@ -29,9 +29,9 @@
 # Checks for header files.
 
 dnl only unix builds should need this stuff, so why...?
-PKG_CHECK_MODULES(_abi_gda,[libgda >= 1.2.0 libgnomedb >= 1.2.0],[
+PKG_CHECK_MODULES(_abi_gda,[libgda-3.0 libgnomedb-extra-3.0],[
 	ABI_PLUGIN_REPORT([gda: okay])
-],[	ABI_PLUGIN_REPORT([gda: error - libgda and libgnomedb >= 1.2.0 required])
+],[	ABI_PLUGIN_REPORT([gda: error - libgda and libgnomedb 3.0.x required])
 	abi_plugin_disable=yes
 ])
 if test $abi_plugin_disable = no; then
--- origsrc/abiword-plugins-2.6.4/tools/gda/unix/AbiGDA.cpp	2008-07-13 07:04:53.000000000 -0500
+++ src/abiword-plugins-2.6.4/tools/gda/unix/AbiGDA.cpp	2008-07-16 16:22:02.843750000 -0500
@@ -51,7 +51,7 @@
 
 #include <gtk/gtk.h>
 #include <libgda/libgda.h>
-#include <libgnomedb/gnome-db-editor.h>
+#include <libgnomedb-extra/gnome-db-editor.h>
 #include <libgnomedb/gnome-db-login.h>
 #include <libgnomedb/gnome-db-util.h>
 
@@ -196,9 +196,9 @@
 		m_row++; m_col = 0;
 	}
 
-	void addCell(const GdaValue *value)
+	void addCell(const GValue *value)
 	{
-		gchar * str = gda_value_stringify (const_cast<GdaValue*>(value));	       
+		gchar * str = gda_value_stringify (const_cast<GValue*>(value));	       
 		addCell (str);
 		g_free (str);
 	}       	
@@ -276,6 +276,7 @@
 	GdaConnection* cnc;
 	GdaDataModel*  recset;
 	GdaCommand*    cmd;
+	GError*        error;
 
 	dsn_name  = gnome_db_login_get_dsn (GNOME_DB_LOGIN(login));
 	user_name = gnome_db_login_get_username (GNOME_DB_LOGIN(login));
@@ -285,13 +286,13 @@
 	sql = editable_txt;
 	g_free (editable_txt);
 	
-	cnc = gda_client_open_connection (connection_pool, dsn_name.utf8_str(), user_name.utf8_str(), password.utf8_str(), GDA_CONNECTION_OPTIONS_READ_ONLY);
+	cnc = gda_client_open_connection (connection_pool, dsn_name.utf8_str(), user_name.utf8_str(), password.utf8_str(), GDA_CONNECTION_OPTIONS_READ_ONLY, &error);
 	if (!GDA_IS_CONNECTION (cnc))
 		return false;
 	
 	/* execute command */
 	cmd = gda_command_new (sql.utf8_str(), GDA_COMMAND_TYPE_SQL, (GdaCommandOptions)0);
-	recset = gda_connection_execute_single_command (cnc, cmd, NULL);
+	recset = gda_connection_execute_select_command (cnc, cmd, NULL, &error);
 	gda_command_free (cmd);
 	
 	if (recset == NULL || !GDA_IS_DATA_MODEL (recset))
@@ -369,7 +370,7 @@
 	char *argv[2];
 
 	/* run gnome-database-properties config tool */
-	argv[0] = (char *) "gnome-database-properties";
+	argv[0] = (char *) "gnome-database-properties-3.0";
 	argv[1] = NULL;
 
 	return g_spawn_async (NULL, argv, NULL, G_SPAWN_SEARCH_PATH,
--- origsrc/abiword-plugins-2.6.4/tools/gdict/unix/GdictAbiPlugin.cpp	2008-07-13 07:04:53.000000000 -0500
+++ src/abiword-plugins-2.6.4/tools/gdict/unix/GdictAbiPlugin.cpp	2008-07-16 16:22:02.875000000 -0500
@@ -53,10 +53,9 @@
   if ( (pid = fork ()) == 0 )
     {
       // child
-      const char *args[3];
-      args[0] = "--noapplet"; // definitely needed for kde users, for example
-      args[1] = search;
-      args[2] = 0;
+      const char *args[2];
+      args[0] = search;
+      args[1] = 0;
       
       exit ( execvp ( "gnome-dictionary", (char **) args ) );
       // TODO: be smarter with warnings and such
--- origsrc/abiword-plugins-2.6.4/wp/impexp/OpenDocument/GNUmakefile.am	2008-07-13 07:05:44.000000000 -0500
+++ src/abiword-plugins-2.6.4/wp/impexp/OpenDocument/GNUmakefile.am	2008-07-16 16:22:02.890625000 -0500
@@ -8,7 +8,7 @@
 
 plugin_LTLIBRARIES = libAbiOpenDocument.la
 
-libAbiOpenDocument_la_SOURCES =	dummy.c
+libAbiOpenDocument_la_SOURCES =	dummy.cc
 
 libAbiOpenDocument_la_LIBADD = @ABI_LIBS@ \
 	common/xp/libAbiOpenDocument_common.la \
--- origsrc/abiword-plugins-2.6.4/wp/impexp/OpenDocument/dummy.cc	1969-12-31 18:00:00.000000000 -0600
+++ src/abiword-plugins-2.6.4/wp/impexp/OpenDocument/dummy.cc	2008-07-16 16:22:02.953125000 -0500
@@ -0,0 +1,3 @@
+// It's just a workaround
+// Libtool needs some source file on libAbiOpenDocument_la_SOURCES or it will give an error while making.
+void dummy_function() {}
--- origsrc/abiword-plugins-2.6.4/wp/impexp/OpenXML/GNUmakefile.am	2008-07-13 07:05:28.000000000 -0500
+++ src/abiword-plugins-2.6.4/wp/impexp/OpenXML/GNUmakefile.am	2008-07-16 16:22:02.984375000 -0500
@@ -8,7 +8,7 @@
 
 plugin_LTLIBRARIES = libAbiOpenXML.la
 
-libAbiOpenXML_la_SOURCES =	dummy.c
+libAbiOpenXML_la_SOURCES =	dummy.cc
 
 libAbiOpenXML_la_LIBADD = @ABI_LIBS@ \
 	common/xp/libAbiOpenXML_common.la \
--- origsrc/abiword-plugins-2.6.4/wp/impexp/OpenXML/dummy.cc	1969-12-31 18:00:00.000000000 -0600
+++ src/abiword-plugins-2.6.4/wp/impexp/OpenXML/dummy.cc	2008-07-16 16:22:03.031250000 -0500
@@ -0,0 +1,4 @@
+/* It's just a workaround
+   Libtool needs some source file on libAbiOpenXML_la_SOURCES or it will give an error while making.
+*/
+void dummy_function() {}
